File: ClassPrevious/class/99/Hw3/vonluh/ParentObject.java
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    January 1998
//
//////////////////////////////////////////////////////////////////////////

//package oglr;

/**
  * ParentObject - Holds time marker for visualization pipeline
  *
  */
public class ParentObject 
{
	/** Holds the time of the object */
	long objectTime;


	/** Constructor  simply sets the time of creation, all derived
	  * classes automatically inherit this constructor
	  */
	public ParentObject()
	{
		objectTime = System.currentTimeMillis();
	}


	/** Methods to Get/Update time of pipeline objects, used when object
	  * is changed in some way.
	  */
	public long getTime() 
	{ 
		return objectTime; 
	}


	/** Methods to Get/Update time of pipeline objects, used when object
	  * is changed in some way.
	  */
	public void updateTime() 
	{ 
		objectTime = System.currentTimeMillis(); 
	}

	
	//
	//
	//
	public boolean doINeedUpdate(long objectTime)
	{
		
		System.out.println("PARENT OBJECT time:" + getTime());
		if (this.getTime() > objectTime ) 
		 	 return true; 
		return false; 
 
	}
	

	/** Methods to print out time of object
	  * 
	  */
	public void printTime() 
	{ 
		//System.out.println ("Print Time: " + objectTime);
	}

	

}

File: WebGL/OGLRen/OGLRen/ParentObject.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * Abstract Cell class which defines the API for all drawable
  * entities to follow
  */

OGLRen.ParentObject = function()
{
	/** Constructor  simply sets the time of creation, all derived
	  * classes automatically inherit this constructor
	  */
	var objectTime = Date.now();


	/** Methods to Get/Update time of pipeline objects, used when object
	  * is changed in some way.
	  */
	this.getTime = function() { return objectTime; }
	/** Methods to Get/Update time of pipeline objects, used when object
	  * is changed in some way.
	  */
	this.updateTime = function() { objectTime = Date.now(); }
}

// Convenience function for inheritance
Function.prototype.inheritsFrom = function( parentClassOrObject ){ 
	if ( parentClassOrObject.constructor == Function ) 
	{ 
		//Normal Inheritance 
		this.prototype = new parentClassOrObject;
		this.prototype.constructor = this;
		this.prototype.parent = parentClassOrObject.prototype;
	} 
	else 
	{ 
		//Pure Virtual Inheritance 
		this.prototype = parentClassOrObject;
		this.prototype.constructor = this;
		this.prototype.parent = parentClassOrObject;
	} 
	return this;
} 